### DISCUSSION TITLE
TOC active on scroll not working?

### DISCUSSION THREAD

**Discussion by galaczi** [skeletonlabs/skeleton#2158] [General]
I am using the TOC component without AppShell, so my scroll target should be the defaut ``body``. However the active on scroll functionality doesn't work. No error messages or a hint why either.  Has anyone experienced this?

I have activeClass set up and working when I click on a link.

**Comment by endigo9740** [comment]
FYI the default value for `scrollTarget` is `body`, so you can remove that if you're providing this in the `use:tocCrawler` settings:
https://github.com/skeletonlabs/skeleton/blob/master/packages/skeleton/src/lib/utilities/TableOfContents/crawler.ts#L23

The goal of the scroll target is to identify the scrollable element surrounding the headings. Can you double check that the `<body>` element is the scrollable element for your app. There's lots of ways to setup a custom page layout, so I just want to confirm.

Can you also confirm the following:

1. Are you using headings H2-h6? (the defaults)
2.  Does the visible list of links populate in the ToC component?
3. If you inspect a heading, do you see an `id` value generated per heading? (handled by the Crawler action)
4. Alternatively, are you setting your own ids manually?

Of course a link to your repo (if it's public) or a minimal reproduction is always welcome for issues like this.

**Reply by galaczi** [reply]
Hi @endigo9740,

1. Yes, h2-h6
2. The links populate as expected
3. ids are generated
4. I am not generating my own ids

I can click on the links and the page scrolls to the correct position, so everything works fine except the active on scroll functionality (automatically changing the active link when scrolling).

My project is not public but I will make a minimal reproduction.

**Reply by endigo9740** [reply]
I see, so basically everything is working, it's just not highlighting the top-most header present on the scrollable region of the page. Is that correct?

I'll look forward to checking out the reproduction. FYI I'm just heading out the door for the day, tomorrow is going to be busy for me, but I will have more time Friday. If someone else doesn't jump in between now and then, I'll plan to test it then. In the meantime I will aim to at least do a quick code review to see if anything stands out.

Thanks!

**Comment by galaczi** [comment]
> I see, so basically everything is working, it's just not highlighting the top-most header present on the scrollable region of the page. Is that correct?

That is correct. 

A big caveat for me that I just realized is that it's a client side only component. I will have to look for an alternative or roll my own.

**Reply by Sarenor** [reply]
Now, I may not quite get what you're tying to do, but aren't anchor links always just clientside? So how should the component work serverside? Or do you mean that it doesn't work without JS clientside?

**Reply by galaczi** [reply]
I mean that the ids and links are generated client side. It's fine from a UX perspective but bad for SEO and I am working on a project that is very particular about this stuff. Two other popular third party components also work client side only so there may be other considerations I am not aware of. I will try to code something custom.

**Reply by endigo9740** [reply]
Yeah, I think it would take a much more sophisticated approach to move this server side. If you find a solution that works well, feel free to share. We're always down for new ideas.
